src/libs/ajax.js

```diff
@@ -76,10 +76,13 @@ const checkRequesterPaysError = async response => {
  * Detects errors due to requester pays buckets, and adds the current workspace's billing
  * project if the user has access, retrying the request once if necessary.
  */
-const withRequesterPays = wrappedFetch => (url, ...args) => {
+const withRequesterPays = wrappedFetch => async (url, ...args) => {
   const bucket = /\/b\/([^/?]+)[/?]/.exec(url)[1]
   const workspace = workspaceStore.get()
-  const userProject = workspace && Utils.canWrite(workspace.accessLevel) ? workspace.workspace.namespace : requesterPaysProjectStore.get()
+  const userProjects = await Billing().listProjects()
+  const userProject = workspace && _.some(['projectName', workspace.workspace.namespace], userProjects) ?
+    workspace.workspace.namespace :
+    requesterPaysProjectStore.get()
```

Comment:
I chose to not write this back to `requesterPaysProjectStore` because that only holds a single project and is only cleared on sign-out. I think it would be better as a map from bucket to project. Otherwise, this bug could be encountered again by, in a single session, viewing two requester pays buckets in two different billing projects.

Reply:
This store is also cleared when navigating from one workspace to another (see `auth.js`). FWIW the originally intended behavior is that once the user selects a billing project, that project is used for all requester pays operations made from the context of that workspace, until they navigate away. Does that change your perspective on writing back to the store?

Reply:
Yes, it does. I saw `auth.js` and incorrectly assumed that it was being cleared on sign-out. I am okay with writing to `requesterPaysProjectStore` here.

That brings up another issue: That reference in `auth.js` is the only call to `reset()` for `requesterPaysProjectStore`. Are we not clearing this on sign-out? If so, that feels like a bug to me.